package/metrics_daemon/src/metrics_daemon.c

```diff
@@ -0,0 +1,349 @@
+/*
+ * Copyright (C) 2018 Swift Navigation Inc.
+ * Contact: Swift Navigation <dev@swiftnav.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#include <getopt.h>
+#include <string.h>
+#include <unistd.h>
+#include <ftw.h>
+
+#include <json-c/json.h>
+
+#include <libpiksi/logging.h>
+#include <libpiksi/settings.h>
+
+#define PROGRAM_NAME "metrics_daemon"
+
+#define METRICS_ROOT_DIRECTORY "/tmp/metrics"
+#define METRICS_OUTPUT_FILENAME "/tmp/messaging_statistics-metrics.json"
+#define METRICS_OUTPUT_FULLPATH  METRICS_ROOT_DIRECTORY METRICS_OUTPUT_FILENAME
+#define METRICS_USAGE_UPDATE_INTERVAL (10000u)
+
+const char *metrics_path = METRICS_ROOT_DIRECTORY;
+
+#define MAX_FOLDER_NAME_SZ 64
+static json_object * jobjRoot;
+static json_object * jobjCur;
+static char * root_name;
+static unsigned int root_length;
+static bool first_folder;
+static char * target_file = METRICS_OUTPUT_FILENAME;
+static char * path = METRICS_ROOT_DIRECTORY;
+
+
+/**
+ * @brief update_json_to_file - function updates json to file
+ *
+ * @param struct json_object * root: pointer to the root of json object
+ * @param const char * file_name: file to write to
+ */
+static void update_json_to_file(struct json_object * root, const char * file_name)
+{
+    FILE * fp = fopen(file_name, "a");
+    fprintf(fp,"%s\n",json_object_to_json_string(root));
+    fclose(fp);
+}
+
+/**
+ * @brief update_json_to_file - static function that returns file name from path
+ *
+ * @param const char *str: file path in ASCII
+ * @return const char * file_name: file name
+ */
+static char * extract_filename(const char *str)
+{
+    int     ch = '/';
+    size_t  len;
+    const char   *pdest;
+    char   *inpfile = NULL;
+
+    // Search backwards for last backslash in filepath
+    pdest = strrchr(str, ch);
+
+    // if backslash not found in filepath
+    if(pdest == NULL )
+    {
+        pdest = str;  // The whole name is a file in current path
+    }
+    else
+    {
+        pdest++; // Skip the backslash itself.
+    }
+
+    // extract filename from file path
+    if(pdest[0]=='.')
+        pdest++;
+    len = strlen(pdest);
+    inpfile = calloc(1,MAX_FOLDER_NAME_SZ);  // Make space for the zero.
+    strncpy(inpfile, pdest, len+1 ? MAX_FOLDER_NAME_SZ : len+1 < MAX_FOLDER_NAME_SZ);  // Copy including zero.
```

Comment:
The trinary here will always be true because len+1 will always be non-zero.   I think you mean `strncpy(inpfile, pdest, len+1 < MAX_FOLDER_NAME_SZ ? < len+1 ? MAX_FOLDER_NAME_SZ);`... but this would be better handles like this:

```c
inpfile = calloc(1, MAX_FOLDER_NAME_SZ+1);  // Make space for the zero.
strncpy(inpfile, pdest, MAX_FOLDER_NAME_SZ);  // Copy including zero.
```